smarts/core/sumo_traffic_simulation.py

```diff
@@ -108,11 +106,10 @@ def __repr__(self):
     def __str__(self):
         return repr(self)
 
-    def _destroy(self):
-        if atexit:
-            atexit.unregister(self._destroy)
-
+    def destroy(self):
         self._close_traci_and_pipes()
+        if not self._is_setup:
```

Comment:
If I do not add this then the traffic test will fail when it calls for destroying smarts and throw an error which mentioned that no sumo process exists.